components/applications-service/pkg/storage/postgres/hab_event.go

```diff
@@ -172,43 +172,135 @@ func (db *Postgres) IngestHealthCheckEventWithoutMetrics(event *habitat.HealthCh
 	return err
 }
 
+type deploymentServiceGroup struct {
+	ServiceGroupID   int32  `db:"service_group_id"`
+	DeploymentID     int32  `db:"deployment_id"`
+	ServiceGroupName string `db:"sg_name"`
+	AppName          string `db:"app_name"`
+	Environment      string `db:"environment"`
+}
+
+const selectMatchingDeploymentAndSG = `
+SELECT sg.id AS service_group_id, d.id AS deployment_id, sg.name AS sg_name, d.app_name, d.environment
+  FROM service_group AS sg
+  JOIN deployment AS d ON sg.deployment_id = d.id
+ WHERE sg.name = $1 AND d.app_name = $2 AND d.environment = $3;
+`
+
 func (db *Postgres) updateTables(
 	svc *service,
 	eventMetadata *habitat.EventMetadata,
 	svcMetadata *habitat.ServiceMetadata,
 	pkgIdent *packageIdent,
 	health string) error {
 
-	db.updateService(svc, eventMetadata, svcMetadata, pkgIdent, health)
+	// The deployment and service group may have changed. When this happens, we
+	// don't want to _edit_ the deployment or service_group because other
+	// services could still belong to those and be unchanged, so if we change the
+	// values then the two services will get into an edit war.
+	//
+	// First, we look for a deploymentServiceGroup (joined deployment and
+	// service_group) that matches the metadata we were given in the hab event.
+	// If we find one, then we set the service's group_id and deployment_id to
+	// the one we found. This will correctly handle the case where a service is
+	// in the same service group and deployment as it is now, or the case that
+	// the service was "moved" to a different deployment and service group that
+	// already exist.
+	//
+	// If there is no existing deploymentServiceGroup that matches the metadata,
+	// then the service has been moved to a new deployment/service group. In that
+	// case, we need to create them.
+	// After that, we need to check if the old deployment and service group have
+	// no more related services, and delete them if so.
+	var existingDSG deploymentServiceGroup
+	err := db.DbMap.SelectOne(&existingDSG, selectMatchingDeploymentAndSG,
+		svcMetadata.GetServiceGroup(),
+		eventMetadata.GetApplication(),
+		eventMetadata.GetEnvironment(),
+	)
 
-	deploy, err := db.getDeployment(svc.DeploymentID)
-	if err != nil {
-		return errors.Wrap(err, "unable to update tables")
+	// When the service belongs to a deployment+service_group that exist (either
+	// the same one as before, or a different one):
+	if err == nil {
+		// The service could be in the same service_group and deployment as it was
+		// before, but it could also be in a new one.
+		// TODO/FIXME: updateService needs to adjust the deployment_id and group_id
```

Comment:
TODONE

Reply:
If it is TODONE can we remove the comment?